enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,472 @@
+detailing that there was no capacity for the instance to be created.
+Users may delete the Machine and attempt to recreate it later.
+
+If using the Cluster Autoscaler, the Cluster Autoscaler will detect the failed launch and will, after some period (15
+minutes by default), attempt to scale a different group of instances to fulfil the capacity gap, if it is able to.
+
+We recommend that users leverage multiple MachineSets, each with their own Placement Group to minimise the likelihood
+of exhausting the capacity of the Placement Group.
+For example, in batch scenarios, users may wish to create a MachineSet per group of workloads. By separating each job
+into their own Placement Group, the number of instances per Placement Group may be kept small while keeping the
+benefits of grouping the workload placement in a cluster.
+
+Having spoken with AWS about the possible limitations, an engineer responded with:
+
+> The maximum number of instances in a placement group with a cluster strategy mainly depends on how many EC2 instances
+are available and close together in the same availability zone. In the past, I have been able to use more than 1,000
+c5.24xlarge instances in the same placement group with a cluster strategy.
+
+This implies that this risk may not actually be that great and that users may not hit the issues we have proposed here.
+
+#### Spread placement groups limit MachineSets to seven instances
+
+Currently, a MachineSet is limited to acting across a single Availability Zone within AWS.
+As a Spread placement group has a limit of seven instances per AZ, this then limits each MachineSet to 7 instances when
+using a Spread placement group.
+
+This may cause issues for autoscaling if users do not configure their autoscaling limits correctly.
+For example, a user may specify their MachineAutoscaler to 8 instances, even though we know it will be impossible for
+the 8th instance to launch.
+In this scenario, were the autoscaler to scale the MachineSet to 8 instances, the final instance would never launch and
+eventually the autoscaler should (dependent on configuration) scale the MachineSet back down and, if possible, attempt
+a different MachineSet.
+
+Unfortunately, there isn't currently a good way to signal to the autoscaler that this limitation exists, nor is there a
+good way to signal back to the end user that the reason for the launch failures is the seven instance limit.
+
+Using existing webhooks that inspect the Machine provider configuration, we can warn users if they create a MachineSet
+with a Spread placement group and attempt to scale it over seven instances.
+This however is not fool-proof, as the user may not have set the placement group type, or may have set it incorrectly
+and be using an existing placement group not created by MAPI.
+In this scenario the warning would not make sense, for this reason, we cannot make the warning a blocking error.
```

Comment:
another option here, would be for the cluster autoscaler operator to expose validating webhooks for the MachineAutoscaler resource which could then check to see if placement groups are being used, and if so check the maximum node group size. if the max size is over 7, it could give a quick response error back to the user.